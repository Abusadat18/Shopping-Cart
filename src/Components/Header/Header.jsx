import React from 'react'
import { Link, NavLink } from 'react-router-dom'
import styles from "./Header.module.css"

function Header() {
  return (
    <header className={styles.header}>
      <div className={styles.top}>
        <h1>BAZAAR PLAZA</h1>
        <div className={styles.actions}>
          <Link><img src="/account-circle.svg" alt="" width={35} /></Link>
          <NavLink to="myCart"><img src="/cart.svg" alt="" width={35} /></NavLink>
        </div>
      </div>
      <nav>
        <NavLink to="/" className={styles.navLink}>Home</NavLink>
        <NavLink to="shop" className={styles.navLink}>Shop</NavLink>
      </nav>
    </header>
  )
}

export default Header