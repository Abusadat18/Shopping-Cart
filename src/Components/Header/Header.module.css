.header {
    display: flex;
    flex-direction: column;
}

.top {
    display: flex;
    justify-content: space-between;
    font-family: titleFont, Arial, Helvetica, sans-serif;
}

.actions {
    display: flex;
    gap: 10px;
}

nav {
    display: flex;
    justify-content: center;
    gap: 20px;
}

.navLink {
    font-size: 18px;
}