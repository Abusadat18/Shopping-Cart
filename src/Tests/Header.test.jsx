import { describe, expect, test } from "vitest";
import Header from "../Components/Header/Header";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("Header Component", () => {
    test("render header", () => {
        const { container } = render(
            <MemoryRouter>
                <Header />
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    });
});