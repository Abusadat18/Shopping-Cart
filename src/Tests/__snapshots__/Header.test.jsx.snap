// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > render header 1`] = `
<div>
  <header
    class="_header_e88af3"
  >
    <div
      class="_top_e88af3"
    >
      <h1>
        BAZAAR PLAZA
      </h1>
      <div
        class="_actions_e88af3"
      >
        <a
          href="/"
        >
          <img
            alt=""
            src="/account-circle.svg"
            width="35"
          />
        </a>
        <a
          class=""
          href="/myCart"
        >
          <img
            alt=""
            src="/cart.svg"
            width="35"
          />
        </a>
      </div>
    </div>
    <nav>
      <a
        aria-current="page"
        class="_navLink_e88af3 active"
        href="/"
      >
        Home
      </a>
      <a
        class="_navLink_e88af3"
        href="/shop"
      >
        Shop
      </a>
    </nav>
  </header>
</div>
`;
